backend/endpoints/rooms.py

```diff
@@ -170,3 +170,95 @@ async def join_room(
         status_code=status.HTTP_424_FAILED_DEPENDENCY,
         detail="failed to add new members to room",
     )
+
+
+@router.put(
+    "/org/{org_id}/rooms/{room_id}",
+    response_model=ResponseModel,
+    status_code=status.HTTP_200_OK,
+    responses={
+        401: {"detail": "member not in room"},
+        403: {"detail": "cannot close a channel room"},
+        404: {"detail": "room not found"},
+        424: {"detail": "unable to close conversation"},
+    },
+)
+async def close_conversation(
+    org_id: str,
+    room_id: str,
+    member_id: str,
+    background_tasks: BackgroundTasks,
+):
+    """
+    Closes a DM or Group_DM room on the sidebar
+    By toggling the closed boolean field of the room document from False to True
+    The function when called on a closed room, changes the closed field back to False
+    Args:
+        org_id (str): A unique identifier of an organisation
+        room_id: A unique identifier of the room to be updated
+        member_id: A unique identifier of the member initiating the request
+        background_tasks: A parameter that allows tasks to be performed outside of the main function
+    Returns:
+        HTTP_200_OK: {
+                        "status": "success",
+                        "message": "conversation closed || conversation opened",
+                        "data": {
+                            "closed": true || false,
+                            "role": "admin",
+                            "starred": false
+                        }
+                    }
+    Raises:
+        HTTP_401_UNAUTHORIZED: member not in room
+        HTTP_403_FORBIDDEN: cannot close a channel room
+        HTTP_404_NOT_FOUND: room not found
+        HTTP_424_FAILED_DEPENDENCY: failed to add new members to room
+    """
+    DB = DataStorage(org_id)
+    room = await get_room(org_id=org_id, room_id=room_id)
+
+    if not room:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="room not found"
+        )
+
+    if room["room_type"].upper() == RoomType.CHANNEL:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN, detail="cannot close a channel room"
+        )
+
+    if member_id not in room["room_members"].keys():
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED, detail="member not in room"
+        )
+
+    member_room_data = room["room_members"].get(member_id)
+    member_room_data["closed"] = member_room_data["closed"] is False
```

Comment:
@mykie88 boss, can u pls explain this line to me?

Reply:
okay boss, member_room_data["closed"] is set to "False" by default, so instead of writing too many lines of code just to change it to "True", I decided to just ask if it is False ("member_room_data["closed"] is False") which will automatically return a true and that will be assigned to the "closed" value. 
The same logic will be applied in the reversed order if the closed value of a room is True, so that a previously closed room can be reopened.
Please let me know if it is clear @man-of-mind

Reply:
We could also write :
member_room_data["closed"] = not member_room_data["closed"]

Reply:
I got confused by this line too.
Thanks for the clarification @mykie88 

dope logic!

Reply:
> okay boss, member_room_data["closed"] is set to "False" by default, so instead of writing too many lines of code just to change it to "True", I decided to just ask if it is False ("member_room_data["closed"] is False") which will automatically return a true and that will be assigned to the "closed" value. 
> The same logic will be applied in the reversed order if the closed value of a room is True, so that a previously closed room can be reopened.
> Please let me know if it is clear @man-of-mind 

Thanks boss @mykie88